### PR TITLE
#165412927 Manage article tags

### PR DESCRIPTION
**Pull request by djallas**
#### What does this PR do?
Users should be to tag their articles

#### Description of Task to be completed?
- Users can be able to tag their articles
- User can be able to delete a tag their created
- Get a list of all tags across all articles

#### Endpoints
- **/PUT** /articles/:slug/tags
- **/DELETE** /articles/:slug/tags
- **/GET** /tags

#### How should this be manually tested?
Use Postman to test tag endpoints

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#165412927](https://www.pivotaltracker.com/story/show/165412927)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

**Comment by e-liyai** [comment]
@djallas I see files I have already merged in your previous PR and are still reported as new files. is this the case?

**Comment by djallas** [comment]
@e-liyai can you let me know which ones are new? because I introduced the files that manage **tags** only, So I want to double check to make sure everything is okay

**Comment by djallas** [comment]
@e-liyai I have evaluated new files. there are marked as new because they are being compared to the previous *develop* branch status before merging with #21. However, they are the same.

**Comment by djallas** [comment]
@e-liyai  Conflict solved

File: src/queries/tags/update.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+export default async (newTags, slug, action) => {
+  let result = [];
+  let message = '';
+  const response = await db.Article.findAll({
+    limit: 1,
+    where: { slug },
+    attributes: ['tagList'],
+    logging: false
+  });
```
Comment (sengayire) [inline src/queries/tags/update.js]: I can see that you find an article by slug not by id, so why

File: src/tests/controllers/TagController.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+const { expect } = chai;
+chai.should();
+
+const article = Factory.article.build();
+chai.use(chaiHttp);
+const token = 'token-string';
+describe('TAGS', () => {
+  let response = '';
+  before(async () => {
+    const findUser = await db.User.findAll({ limit: 1, logging: false });
```
Comment (sengayire) [inline src/tests/controllers/TagController.test.js]: you find all from User model and you limit is 1, you can use findOne instead

File: src/queries/tags/update.js (hunk 1 above)
Comment (djallas) [reply src/queries/tags/update.js]: As it is mentioned in the app **README** and **PT Stories**,  the identifier of the article is `slug` not `id`

File: src/tests/queries/tags/delete.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import * as Factory from '../../../helpers/factory';
+import db from '../../../models';
+
+const article = Factory.article.build();
+delete article.id;
+chai.use(chaiHttp);
+describe('Tag query', () => {
+  it('should update tag', async () => {
+    const action = 'detele';
+    const findArticle = await db.Article.findAll({ limit: 1, logging: false });
```
Comment (sengayire) [inline src/tests/queries/tags/delete.test.js]: And here

File: src/queries/tags/update.js (hunk 1 above)
Comment (MBenedicte) [reply src/queries/tags/update.js]: I think that has been done.. you did well to create another file

File: src/tests/controllers/TagController.test.js (hunk 2 above)
Comment (djallas) [reply src/tests/controllers/TagController.test.js]: With **findAll**, it is easy to return one format of the response to the controller

File: src/tests/queries/tags/delete.test.js (hunk 3 above)
Comment (djallas) [reply src/tests/queries/tags/delete.test.js]: With **findAll**, it is easy to return one format of the response to the controller

**Comment by denislohan** [comment]
error handling is poor : 
improve it to tackle all possible scenarios.
improve your messaging to be specific
Let the feature be reasonable: eg: an empty string tag is not a tag

**Pull request merged**